### PR TITLE
Revert "Add optimizer rule to move a limit beneath a union operator"

### PR DESCRIPTION
**Pull request by mkleen**
This reverts commit bc0c84f69005a7240fd424c4b3a418aa22683f3b.

Resolves https://github.com/crate/crate/issues/15294

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/planner/PlanPrinterTest.java [hunk 1]
```diff
@@ -100,31 +100,29 @@ public void testUnion() {
...
-                "type=executionPlan}}"));
+               "left={Collect={" +
+                    "collectPhase={COLLECT={distribution={distributedByColumn=0, type=BROADCAST}, " +
+                    "executionNodes=[n1], id=0, orderBy=x ASC, " +
+                    "routing={n1={t1=[0, 1, 2, 3]}}, " +
+                    "toCollect=[x], " +
+                    "type=executionPhase, " +
+                    "where=true}}, " +
+                    "type=executionPlan}}, " +
+               "mergePhase={MERGE={" +
```
Comment (matriv) [inline server/src/test/java/io/crate/planner/PlanPrinterTest.java:113]: needs one more indent step?

**Review by matriv — APPROVED** [review]
thx!

File: server/src/test/java/io/crate/planner/PlanPrinterTest.java (hunk 1 above)
Comment (mkleen) [reply server/src/test/java/io/crate/planner/PlanPrinterTest.java:113]: This commit is a just reverting to the previous state of the file.

Comment (matriv) [reply server/src/test/java/io/crate/planner/PlanPrinterTest.java:113]: cool cool, thx!

**Pull request merged**